### ISSUE TITLE
Issue with JIT

### ISSUE THREAD

**Issue by vinayakkulkarni** [nuxt-community/tailwindcss-module#320]
### Version
@nuxtjs/tailwindcss: v4.0.2
nuxt: v2.15.3
node: v15.12.0
npm: v7.6.3

### Reproduction Link
N/A. Will try and add soon.

### Steps to reproduce
Enable JIT in `nuxt.config.ts`
```ts
tailwindcss: {
  cssPath: '~/assets/css/tailwind.css',
  configPath: 'tailwind.config.js',
  exposeConfig: false,
  config: {},
  viewer: false,
  jit: true,
},
```

### What is Expected?
Jit to work

### What is actually happening?
```sh
 WARN  postcss@8.2.8 is installed but ^7.0.32 is expected                                                                                                                                                                                                           12:55:42

ℹ Using Tailwind CSS from ~/assets/css/tailwind.css                                                                                                                                                                                                nuxt:tailwindcss 12:55:43
ℹ Merging Tailwind config from ~/tailwind.config.js                                                                                                                                                                                                nuxt:tailwindcss 12:55:43
ℹ Sentry reporting is disabled ("disabled" option has been set)                                                                                                                                                                                         nuxt:sentry 12:55:43

   ╭───────────────────────────────────────╮
   │                                       │
   │   Nuxt @ v2.15.3                      │
   │                                       │
   │   ▸ Environment: development          │
   │   ▸ Rendering:   client-side          │
   │   ▸ Target:      static               │
   │                                       │
   │   Listening: http://localhost:3000/   │
   │                                       │
   ╰───────────────────────────────────────╯

ℹ Preparing project for development                                                                                                                                                                                                                                 12:55:46
ℹ Initial build may take a while                                                                                                                                                                                                                                    12:55:46
ℹ Tailwind JIT activated                                                                                                                                                                                                                           nuxt:tailwindcss 12:55:46
✔ Builder initialized                                                                                                                                                                                                                                               12:55:46
✔ Nuxt files generated                                                                                                                                                                                                                                              12:55:46

✖ Client
  Compiled with some errors in 40.25s


 ERROR  Failed to compile with 1 errors                                                                                                                                                                                                             friendly-errors 12:56:31


 ERROR  in ./.nuxt/components/nuxt-error.vue?vue&type=style&index=0&lang=css&                                                                                                                                                                       friendly-errors 12:56:31

Module build failed (from ./node_modules/@nuxt/postcss8/node_modules/postcss-loader/dist/cjs.js):                                                                                                                                                   friendly-errors 12:56:31
TypeError: Cannot read property 'type' of null
    at extractCandidates (/Users/vinayak/Development/Work/project/abc/node_modules/@tailwindcss/jit/src/lib/setupContext.js:314:22)
    at /Users/vinayak/Development/Work/project/abc/node_modules/@tailwindcss/jit/src/lib/setupContext.js:328:22
    at Array.flatMap (<anonymous>)
    at withIdentifiers (/Users/vinayak/Development/Work/project/abc/node_modules/@tailwindcss/jit/src/lib/setupContext.js:326:30)
    at addBase (/Users/vinayak/Development/Work/project/abc/node_modules/@tailwindcss/jit/src/lib/setupContext.js:390:38)
    at /Users/vinayak/Development/Work/project/abc/node_modules/@tailwindcss/forms/src/index.js:252:5
    at registerPlugins (/Users/vinayak/Development/Work/project/abc/node_modules/@tailwindcss/jit/src/lib/setupContext.js:580:7)
    at /Users/vinayak/Development/Work/project/abc/node_modules/@tailwindcss/jit/src/lib/setupContext.js:770:5
    at /Users/vinayak/Development/Work/project/abc/node_modules/@tailwindcss/jit/src/index.js:37:49
    at LazyResult.runOnRoot (/Users/vinayak/Development/Work/project/abc/node_modules/postcss/lib/lazy-result.js:303:16)
    at LazyResult.runAsync (/Users/vinayak/Development/Work/project/abc/node_modules/postcss/lib/lazy-result.js:355:26)
    at async Object.loader (/Users/vinayak/Development/Work/project/abc/node_modules/@nuxt/postcss8/node_modules/postcss-loader/dist/index.js:95:14)
                                                                                                                                                                                                                                                    friendly-errors 12:56:31
 @ ./node_modules/vue-style-loader??ref--3-oneOf-1-0!./node_modules/@nuxt/postcss8/node_modules/css-loader/dist/cjs.js??ref--3-oneOf-1-1!./node_modules/vue-loader/lib/loaders/stylePostLoader.js!./node_modules/@nuxt/postcss8/node_modules/postcss-loader/dist/cjs.js??ref--3-oneOf-1-2!./node_modules/vue-loader/lib??vue-loader-options!./.nuxt/components/nuxt-error.vue?vue&type=style&index=0&lang=css& 4:14-380 15:3-20:5 16:22-388
 @ ./.nuxt/components/nuxt-error.vue?vue&type=style&index=0&lang=css&
 @ ./.nuxt/components/nuxt-error.vue
 @ ./.nuxt/index.js
 @ ./.nuxt/client.js
 @ multi ./node_modules/eventsource-polyfill/dist/browserify-eventsource.js (webpack)-hot-middleware/client.js?reload=true&timeout=30000&ansiColors=&overlayStyles=&path=%2F__webpack_hmr%2Fclient&name=client ./.nuxt/client.js
```

Workaround
```diff
{
  cssPath: '~/assets/css/tailwind.css',
  configPath: 'tailwind.config.js',
  exposeConfig: false,
  config: {},
  viewer: false,
-  jit: true,
+  jit: false,
}
```

**Comment by vinayakkulkarni** [comment]
OK, one of the dependabot update was failing ci,

fix(deps): bump @tailwindcss/forms from 0.2.1 to 0.3.0

Tailwind Forms plugin has issues with jit?

**Comment by vinayakkulkarni** [comment]
- open bug in jit repo - https://github.com/tailwindlabs/tailwindcss-jit/issues/158

**Comment by vinayakkulkarni** [comment]
Issue is fixed by upgrading `@tailwindcss/form` to `v0.3.2`

Reference - https://github.com/tailwindlabs/tailwindcss-jit/issues/158#issuecomment-808117138

**Comment by kgnfth** [comment]
i have this error with and without forms plugin and with blank tailwind config

```python
 ERROR  Failed to compile with 1 errors


 ERROR  in ./node_modules/@nuxtjs/tailwindcss/dist/runtime/tailwind.css

Module build failed (from ./node_modules/@nuxt/postcss8/node_modules/postcss-loader/dist/cjs.js):
RangeError: Maximum call stack size exceeded
    at candidatePermutations (/home/deejavu/nuxtravolta/node_modules/@tailwindcss/jit/src/lib/generateRules.js:19:32)
    at candidatePermutations (/home/deejavu/nuxtravolta/node_modules/@tailwindcss/jit/src/lib/generateRules.js:41:10)
    at candidatePermutations.next (<anonymous>)
    at candidatePermutations (/home/deejavu/nuxtravolta/node_modules/@tailwindcss/jit/src/lib/generateRules.js:41:10)
    at candidatePermutations.next (<anonymous>)
    at candidatePermutations (/home/deejavu/nuxtravolta/node_modules/@tailwindcss/jit/src/lib/generateRules.js:41:10)
    at candidatePermutations.next (<anonymous>)
    at candidatePermutations (/home/deejavu/nuxtravolta/node_modules/@tailwindcss/jit/src/lib/generateRules.js:41:10)
    at candidatePermutations.next (<anonymous>)
    at candidatePermutations (/home/deejavu/nuxtravolta/node_modules/@tailwindcss/jit/src/lib/generateRules.js:41:10)
    at candidatePermutations.next (<anonymous>)
    at candidatePermutations (/home/deejavu/nuxtravolta/node_modules/@tailwindcss/jit/src/lib/generateRules.js:41:10)
    at candidatePermutations.next (<anonymous>)
    at candidatePermutations (/home/deejavu/nuxtravolta/node_modules/@tailwindcss/jit/src/lib/generateRules.js:41:10)
    at candidatePermutations.next (<anonymous>)
    at candidatePermutations (/home/deejavu/nuxtravolta/node_modules/@tailwindcss/jit/src/lib/generateRules.js:41:10)

 @ ./node_modules/@nuxtjs/tailwindcss/dist/runtime/tailwind.css 4:14-207 15:3-20:5 16:22-215
 @ ./.nuxt/App.js
 @ ./.nuxt/index.js
 @ ./.nuxt/client.js
 @ multi ./node_modules/eventsource-polyfill/dist/browserify-eventsource.js (webpack)-hot-middleware/client.js?reload=true&timeout=30000&ansiColors=&overlayStyles=&path=%2F__webpack_hmr%2Fclient&name=client ./.nuxt/client.js

ℹ Waiting for file changes
ℹ Memory usage: 252 MB (RSS: 544 MB)
ℹ Listening on: http://localhost:3000/
```
